misk-hibernate/src/test/kotlin/misk/hibernate/MySqlScaleSafetyTest.kt

```diff
@@ -0,0 +1,59 @@
+package misk.hibernate
+
+import misk.jdbc.DataSourceType
+import misk.jdbc.TableScanException
+import misk.jdbc.uniqueLong
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.time.LocalDate
+import javax.inject.Inject
+
+/**
+ * Verifies that we're constraining a few things that makes apps hard to scale out.
+ */
+@MiskTest(startService = true)
+class MySqlScaleSafetyTest {
+  @MiskTestModule
+  val module = MoviesTestModule(DataSourceType.MYSQL)
+
+  @Inject @Movies lateinit var transacter: Transacter
+
+  @Test
+  fun tableScansDetected() {
+    transacter.transaction { session ->
+      session.withoutChecks(Check.FULL_SCATTER, Check.COWRITE) {
```

Comment:
hmm these `withoutChecks` shouldn't need to be here? FULL_SCATTER and COWRITE can only happen on vitess

Reply:
you're right... deleted